execute/internal/gas/evm/gas_helpers.go

```diff
@@ -54,7 +92,28 @@ func (gp EstimateProvider) CalculateMessageMaxGas(msg ccipocr3.Message) uint64 {
 	var data []byte = msg.Data
 	dataLength := len(data)
 
-	// TODO: parse max gas from ExtraArgs.
+	// If msg is an EVM message, its ExtraArgs field is an encoded EVMExtraArgsV2 struct, defined in Client.sol here:
+	//nolint:lll
+	// https://github.com/smartcontractkit/ccip/blob/dee09c782c17de1e37e3a1cf625d430330532c6d/contracts/src/v0.8/ccip/libraries/Client.sol#L47
+	gasLimit, err := decodeEVMExtraArgsV2GasLimit(msg.ExtraArgs)
```

Comment:
can we switch based on the first 4 bytes of selector and use the corresponding extra arg decoder version, as opposed to hardcoded v2; the v2 extra args selector is defined in `clients.sol`: 
```
bytes4 public constant EVM_EXTRA_ARGS_V2_TAG = 0x181dcf10;
```

Reply:
This logic will only handle v2 encoded args though, right? I guess it makes sense to confirm the first 4 bytes are the V2 tag.

> btw shouldn't this be implemented somewhere else to also parse the OOO flag?

I copied from this function in `chainlink`, which also just picks out the gasLimit:
https://github.com/smartcontractkit/chainlink/blob/c1a39ebbf0226d1efafac07034b16625d45b3f82/core/capabilities/ccip/ccipevm/helpers.go#L11